src/data_aggregator.ts

```diff
@@ -107,100 +133,120 @@ export class DataAggregator {
   constructor(config: DataAggregatorConfig) {
     this.config = config
     this.logger = this.config.baseLogger.child({ context: 'data_aggregator' })
-    this.exchangeAdapters = this.setupExchangeAdapters()
+    this.priceSources = this.setupPriceSources()
   }
 
-  private setupExchangeAdapters(): ExchangeAdapter[] {
-    const adapterConfig = {
+  private setupPriceSources(): PriceSource[] {
+    const baseAdapterConfig = {
       apiRequestTimeout: this.config.apiRequestTimeout,
-      baseCurrency: CurrencyPairBaseQuote[this.config.currencyPair].base,
       baseLogger: this.config.baseLogger,
-      quoteCurrency: CurrencyPairBaseQuote[this.config.currencyPair].quote,
       metricCollector: this.config.metricCollector,
     }
-
-    if (!this.config.exchanges) {
-      this.config.exchanges = Object.keys(Exchange).map((e) => e as Exchange)
+    const adapterFactory: AdapterFactory = (
+      exchange: Exchange,
+      pair: OracleCurrencyPair
+    ): ExchangeAdapter => {
+      const config = {
+        ...baseAdapterConfig,
+        baseCurrency: CurrencyPairBaseQuote[pair].base,
+        quoteCurrency: CurrencyPairBaseQuote[pair].quote,
+      }
+      return adapterFromExchangeName(exchange, config)
     }
 
-    // Protect against duplicates
-    this.config.exchanges = [...new Set(this.config.exchanges)]
-    this.logger.info({ exchanges: this.config.exchanges }, 'Setting up exchange adapter set')
+    let priceSourceConfigs = this.config.priceSourceConfigs ?? ([] as ExchangePriceSourceConfig[])
 
-    return this.config.exchanges.map((exchange) => {
-      const adapter = adapterFromExchangeName(exchange, adapterConfig)
-      this.logger.info(
+    // Append any configs passed in via exchange to priceSourceConfigs.
```

Comment:
I wonder if it makes sense to support two different methods of specifying the price sources, or just deprecate `config.exchanges`.

In the long term, it seems confusing to specify them in two different ways. I'm in favor of simplifying the configuration and deprecating the old method. If it's deprecated, we'd have to make sure the configs of all oracle instances get updated to use `priceSourceConfigs` instead. That work seems manageable at this stage.

Are there reasons to keep this around that I'm missing?

Reply:
I left it so that we could have an oracle that worked with both old and new configs as we transition. Since the config is already broken, we can remove this one now. Removed.